Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -1190,11 +1190,16 @@ private extension ChatChannelController {
                 switch result {
                 case let .success(payload):
                     self.state = .remoteDataFetched
+
                     self.updateNewestFetchedMessageId(with: payload)
                     self.updateOldestFetchedMessageId(with: payload)
-                    if let pageSize = self.channelQuery.pagination?.pageSize {
-                        self.hasLoadedAllPreviousMessages = payload.messages.count < pageSize
+
+                    let pageSize = self.channelQuery.pagination?.pageSize ?? .messagesPageSize
+                    self.hasLoadedAllPreviousMessages = payload.messages.count < pageSize
+                    if self.isJumpingToMessage {
+                        self.isJumpingToMessage = payload.messages.count >= pageSize
                     }
```

Comment:
I am not getting this logic 🤔 How would we have more messages than the page size if we are synchronizing?